### PR TITLE
add the ability to map data to headers

### PR DESCRIPTION
**Pull request by RobinMalfait**

### CODE REVIEW

File: src/resources/appointments.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ export const appointments = createResourceDescription(
       'initiated_by',
     ],
     filters: ['state', 'start', 'end'],
+    headers: {
+      recaptcha: value => ({ 'X-Im-Not-A-Robot': value }),
+    },
```
Comment (RobinMalfait) [inline src/resources/appointments.js]: The reason that I made it a function is so that we have the full flexibility.
We can now do the following:

1. Map to multiple headers
2. Format the header value

File: src/resources/appointments.js [hunk 2]
```diff
@@ -18,6 +18,9 @@ export const appointments = createResourceDescription(
       'initiated_by',
     ],
     filters: ['state', 'start', 'end'],
+    headers: {
+      recaptcha: value => ({ 'X-Im-Not-A-Robot': value }),
```
Comment (RobinMalfait) [inline src/resources/appointments.js]: The reason that I made it a function is so that we have the full flexibility.
We can now do the following:

1. Map to multiple headers
2. Format the header value if we need to
3. Conditionally apply headers based on the value

File: src/util/omit.test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import omit from './omit'
+
+describe('omit', () => {
+  it('should skip non objects', () => {
+    expect(omit(undefined)).toBe(undefined)
+    expect(omit(null)).toBe(null)
+    expect(omit('')).toBe('')
```
Comment (gevalo1) [inline src/util/omit.test.js]: Might be useful to also add a test with an (empty) array.

File: src/util/omit.test.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import omit from './omit'
+
+describe('omit', () => {
```
Comment (gevalo1) [inline src/util/omit.test.js]: - Add a test which checks that when trying to omit all values in the object, it returns an empty object.
- Add a test which checks that nothing goes wrong when trying to omit a value which isn't present in the object.
